extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js

```diff
@@ -493,31 +509,41 @@ export class AmpLightboxViewer extends AMP.BaseElement {
    * @return {!Promise}
    */
   open_(element) {
-    this.getViewport().enterLightboxMode();
+    return this.initializeLightboxIfNecessary_().then(() => {
+      this.getViewport().enterLightboxMode();
 
-    toggle(this.element, true);
-    this.active_ = true;
+      toggle(this.element, true);
+      this.active_ = true;
 
-    this.updateInViewport(dev().assertElement(this.container_), true);
-    this.scheduleLayout(dev().assertElement(this.container_));
+      this.updateInViewport(dev().assertElement(this.container_), true);
+      this.scheduleLayout(dev().assertElement(this.container_));
 
-    this.win.document.documentElement.addEventListener(
-        'keydown', this.boundHandleKeyboardEvents_);
+      this.win.document.documentElement.addEventListener(
+          'keydown', this.boundHandleKeyboardEvents_);
 
-    return this.resources_.requireLayout(dev().assertElement(this.carousel_))
-        .then(() => {
-          this.currentElemId_ = element.lightboxItemId;
-          // Hack to access private property. Better than not getting
-          // type checking to work.
-          /**@type {?}*/ (this.carousel_).implementation_.showSlideWhenReady(
-              this.currentElemId_);
-          const tagName = this.elementsMetadata_[this.currentElemId_]
-              .tagName;
-          if (tagName === 'AMP-IMG') {
-            this.resizeImageViewerDimensions_();
-          }
-          this.updateDescriptionBox_();
-        });
+      return this.resources_.requireLayout(dev().assertElement(this.carousel_))
+          .then(() => this.openLightboxForElement_(element));
+    });
+  }
+
+  /**
```

Comment:
Pulled this code out into own function for readability.

Reply:
Nice, yeah it definitely helps.